extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -344,27 +357,42 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
   ): GeoJsonSource = apply {
     onDataParsed?.let { listener ->
       ignoreParsedGeoJsonRegistry[sourceId] = false
-      // remove any events from queue before posting this task
-      workerHandler.removeCallbacksAndMessages(null)
-      workerHandler.post {
-        val property = data.toPropertyValue()
-        mainHandler.post {
-          // we set parsed data when sync setter was not called during background work
-          if (ignoreParsedGeoJsonRegistry[sourceId] == false) {
-            setProperty(property, throwRuntimeException = false)
-            listener.invoke(this)
+      // No parseJob is working now, parse the data directly.
+      if (parsingJob == null) {
+        parseData(data, listener)
+      } else {
+        parsingJob?.let {
+          if (it.isActive) {
+            // still have parse job working, save the current data to waitingData.
+            pendingData = Pair(data, listener)
```

Comment:
we save only one data for future parsing? 
maybe we should save it in list (queue)

Reply:
No need to parse the pending data if have new pending data coming, we only need to parse the last one.